### PR TITLE
Apple recognizer concurrency fixes

### PR DESCRIPTION
**Pull request by noelweichbrodt**
Suggested by https://github.com/spokestack/spokestack-ios/compare/14.0.3...xaphod:xaphod/14.0.3 as fixes for #109 & #110.

Memory usage is now constant, and repeated simulator runs suggest the concurrency issues wrt `AudioEngine` are no longer impacting usage.

Also, some minor code changes and comments to help future debugging in `TextToSpeech`

### CODE REVIEW

File: Spokestack/AppleSpeechRecognizer.swift [hunk 1]
```diff
@@ -36,6 +39,10 @@ import Speech
     // MARK: NSObject implementation
     
     deinit {
+        if recognizingSemaphore.wait(timeout: .now() + self.configuration.semaphoreTimeout) == .timedOut {
+            self.context.dispatch { $0.failure(error: SpeechPipelineError.illegalState("Could not start recognizer, timed out waiting on another thread.")) }
```
Comment (space-pope) [inline Spokestack/AppleSpeechRecognizer.swift:43]: This is a long error message to be copy pasta'd everywhere.

**Review by space-pope — COMMENTED** [review]
Pretty sure I get what we're doing here, but I have a lingering concern: `AppleSpeechRecognizer` and `AppleWakewordRecognizer` use the _same_ external resources to do their thing, and they both rely on a thread pool (of unknown size, afaik) to submit their tasks. As such, in order to eliminate a potential race condition between the two, they'd have to share a semaphore rather than construct their own.

**Comment by noelweichbrodt** [comment]
If you're talking about `SFSpeechAudioBufferRecognitionRequest` or `AVAudioEngine` wrt "use the same external resources to do their thing", then I see what you're concerned about but don't share it. `AVAudioEngine` is a resource graph that may be thought of as a multiplexed audio manager, so each instance operates independently by design. `SFSpeechAudioBufferRecognitionRequest` is just a single request that is expected to operate independently, and is fed audio from `AVAudioEngine`, so multiple requests held by different places may compete for resources, but should not be managed (outside of official guidance I haven't seen) by their instantiators.

File: Spokestack/AppleSpeechRecognizer.swift (hunk 1 above)
Comment (noelweichbrodt) [reply Spokestack/AppleSpeechRecognizer.swift:43]: error message is pretty short, error handling is rather officious

**Comment by space-pope** [comment]
I mean that the device has a single microphone that these classes have to compete for, but the `AVAudioEngine` abstraction does make working with it a little different than in Android-land. Since you're not managing the microphone directly, I'd wager there's a good chance of the ASR losing a frame (or a few) on the activation edge, but that's probably not a big deal.

**Review by space-pope — APPROVED** [review]

**Comment by noelweichbrodt** [comment]
Yeah, there's exactly zero guarantees for real-time delivery of frames, it's all best-effort.

**Pull request merged**